### PR TITLE
TYP: fix mypy ignored err in pandas/io/formats/format.py

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] xref #37715
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Handle ignored mypy error in ``pandas/io/formats/format.py``

### CODE REVIEW

File: pandas/io/formats/format.py
Comment (simonjayhawkins) [inline pandas/io/formats/format.py:1052]:
```suggestion
        if path_or_buf is None:
            created_buffer = True
            path_or_buf = StringIO()
        else:
            created_buffer = False
```

**Review by simonjayhawkins — APPROVED** [review]
Thanks @ivanovmg lgtm although the assert would be redundant with a slight refactor.

**Review by simonjayhawkins — APPROVED** [review]

**Comment by ivanovmg** [comment]
@simonjayhawkins it's green

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins it's green

since this is changing code, will wait for 2nd approval.

**Comment by jbrockmendel** [comment]
LGTM

**Comment by simonjayhawkins** [comment]
Thanks @ivanovmg

**Pull request merged**
